datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -1478,6 +1478,28 @@ macro_rules! to_string {
     }};
 }
 
+
+/// Array_union SQL function
+pub fn array_union(args: &[ArrayRef]) -> Result<ArrayRef> {
```

Comment:
Please add some check here in case of `args` is of unexpected size

Reply:
@comphead the current implementation is buggy, because he doesn't remove duplicates. I tried a couple of ways and after discussing in the arrow-datafusion Slack channel, it seems the simpler approach would be to convert the array to rows and de-duplicate, then back to rows. What do you think?

https://docs.rs/arrow-row/latest/arrow_row/struct.RowConverter.html#method.convert_rows